### PR TITLE
Add mask for nbr

### PR DESCRIPTION
**Pull request by benjaminnNgo**
### Summary / Description

This is to facilitate the communication between hooks, because right now we let the user define the seed node themselves, the orders of neighbour information are now dynamically determined, and there is no way for downstream hooks to know the order.

**Related Issues:** #324

#### Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking Change
- [x] Refactoring
- [ ] Documentation update

#### Test Evidence

Describe how this PR has been tested.

- [x] Unit tests
- [ ] Integration tests
- [ ] Performance tests

### CODE REVIEW

File: examples/nodeproppred/dygformer.py [hunk 1]
```diff
@@ -142,16 +142,19 @@ def forward(
         nbr_nids = batch.nbr_nids[0]
         nbr_times = batch.nbr_times[0]
         nbr_feats = batch.nbr_feats[0]
+        src_nbr_idx = batch.seed_node_nbr_mask['src']
+        dst_nbr_idx = batch.seed_node_nbr_mask['dst']
         edge_idx = torch.stack((src, dst), dim=0)
-        batch_size = src.shape[0]
+        src.shape[0]
```
Comment (Jacob-Chmura) [inline examples/nodeproppred/dygformer.py]:
```suggestion
```

File: tgm/hooks/neighbors.py [hunk 2]
```diff
@@ -119,12 +119,17 @@ def _append_empty_hop() -> None:
             batch.nbr_times.append(nbr_times)  # type: ignore
             batch.nbr_feats.append(nbr_feats)  # type: ignore
 
+        batch.seed_node_nbr_mask = seed_node_nbr_mask  # type: ignore
```
Comment (Jacob-Chmura) [inline tgm/hooks/neighbors.py:123]: If  `not seed_nodes.numel()` triggers (empty batch), then this attribute won't exist on the batch. I think it would be slightly safer to add the empty `seed_node_mask` dict in this case.

So we can just move this up

File: tgm/hooks/neighbors.py [hunk 3]
```diff
@@ -166,6 +171,11 @@ def _get_seed_tensors(self, batch: DGBatch) -> Tuple[torch.Tensor, torch.Tensor]
                             f'got values in range [{tensor.min().item()}, {tensor.max().item()}]'
                         )
                     seeds.append(seed.to(device))
+                    num_seed_nodes = tensor.shape[0]
+                    seed_node_mask[name] = torch.arange(
+                        offset, offset + num_seed_nodes
+                    ).to(device)
```
Comment (Jacob-Chmura) [inline tgm/hooks/neighbors.py]: Nit: 
```suggestion
                    seed_node_mask[name] = torch.arange(offset, offset + num_seed_nodes, device=device)
```

**Review by Jacob-Chmura — CHANGES_REQUESTED** [review]
I like this, just some minor comments before approve and merge

**Review by Jacob-Chmura — APPROVED** [review]
